src/library_webgpu.js

```diff
@@ -958,11 +996,74 @@ var LibraryWebGPU = {
     var pass = WebGPU.mgrRenderPassEncoder.get(passId);
     pass.drawIndexed(indexCount, instanceCount, firstIndex, baseVertex, firstInstance);
   },
+  dawnRenderPassEncoderExecuteBundles: function(passId, count, bundlesPtr) {
+    var pass = WebGPU.mgrRenderPassEncoder.get(passId);
+
+#if ASSERTIONS
+    assert(bundlesPtr % 4 === 0);
+#endif
+
+    var bundles = Array.from(HEAP32.subarray(bundlesPtr >> 2, (bundlesPtr >> 2) + count),
+      function(id) { return WebGPU.mgrRenderBundle.get(id); });
+    pass.executeBundles(bundles);
+  },
   dawnRenderPassEncoderEndPass: function(passId) {
     var pass = WebGPU.mgrRenderPassEncoder.get(passId);
     pass.endPass();
   },
 
+  // Render bundle encoder
+
+  dawnRenderBundleEncoderSetBindGroup: function(bundleId, groupIndex, groupId, dynamicOffsetCount, dynamicOffsetsPtr) {
+    var pass = WebGPU.mgrRenderBundleEncoder.get(bundleId);
+    var group = WebGPU.mgrBindGroup.get(groupId);
+    if (dynamicOffsetCount == 0) {
+      pass.setBindGroup(groupIndex, group);
+    } else {
+      var offsets = [];
+      for (var i = 0; i < dynamicOffsetCount; i++, dynamicOffsetsPtr += 8) {
+        offsets.push({{{ gpu.makeGetU64('dynamicOffsetsPtr', 0) }}});
+      }
+      pass.setBindGroup(groupIndex, group, offsets);
+    }
+  },
+  dawnRenderBundleEncoderSetIndexBuffer: function(bundleId, bufferId, offset) {
+    var pass = WebGPU.mgrRenderBundleEncoder.get(bundleId);
+    var buffer = WebGPU.mgrBuffer.get(bufferId);
+    pass.setIndexBuffer(buffer, offset);
+  },
+  dawnRenderBundleEncoderSetPipeline: function(bundleId, pipelineId) {
+    var pass = WebGPU.mgrRenderBundleEncoder.get(bundleId);
+    var pipeline = WebGPU.mgrRenderPipeline.get(pipelineId);
+    pass.setPipeline(pipeline);
+  },
+  dawnRenderBundleEncoderSetVertexBuffers: function(bundleId, startSlot, count, buffersPtr, offsetsPtr) {
+    var pass = WebGPU.mgrRenderBundleEncoder.get(bundleId);
+
+#if ASSERTIONS
+    assert(buffersPtr % 4 === 0);
+#endif
+    var buffers = Array.from(HEAP32.subarray(buffersPtr >> 2, (buffersPtr >> 2) + count),
+      function(id) { return WebGPU.mgrBuffer.get(id); });
```

Comment:
Could probably just as easily be inside the loop

Reply:
`dawnRenderPassEncoderSetVertexBuffers` is this way too. Do you want to change that one as well?

Reply:
Doesn't matter that much since we'll have to change it when we roll dawn anyway, so either way.

Reply:
Okay. will leave it for now.